File: 2.1P/CounterObject/Counter.cs
namespace CounterObject
{
	public class Counter
	{
		private int _count = 0;
		private string _name;

		public Counter(string name) => _name = name;

		public string Name { get => _name; set => _name = value; }
		public int Ticks => _count;

		public void Increment() => ++_count;
		public void Decrement() => --_count;
		public void Reset() => _count = 0;
	}
}

File: 11.1P/P11_1/src/counter.rs
pub struct Counter {
	count: i32,
	pub name: String,
}

impl Counter {
	pub fn new(name: &str) -> Self {
		Self { count: 0, name: name.into() }
	}

	pub fn inc(&mut self) {
		self.count += 1;
	}

	pub fn dec(&mut self) {
		self.count -= 1;
	}

	pub fn reset(&mut self) {
		self.count = 0;
	}

	pub fn get_count(&self) -> i32 { self.count }
}
